sklearn/feature_selection/from_model.py

```diff
@@ -183,10 +187,23 @@ class SelectFromModel(BaseEstimator, SelectorMixin):
     `threshold_`: float
         The threshold value used for feature selection.
     """
-    def __init__(self, estimator, threshold=None, prefit=False):
+    def __init__(self, estimator, threshold=None, prefit=False,
+                 max_n_features=None):
         self.estimator = estimator
         self.threshold = threshold
         self.prefit = prefit
+        self.max_n_features = max_n_features
+
+    def _check_params(self, X, y):
+        X, y = check_X_y(X, y)
+        if self.max_n_features is not None and not (
+                            self.max_n_features == "all" or
```

Comment:
If it were up to me, I would do 
```
if self.max_n_features is not None:
    if isinstance(self.max_n_features, int):
        # Do check here
    elif isinstance(self.max_n_features, str):
        # Do check here
    raise ValueError("max_n_features must be type str ('all') or int")
```
You could optionally make a `_check_max_n_features` helper function to encapsulate this hairly logic.